### ISSUE TITLE
Unreachable log info of frame_try_runtime::TryRuntime

### ISSUE THREAD

**Issue by Chralt98** [KILTprotocol/kilt-node#362]
### Discussed in https://github.com/KILTprotocol/kilt-node/discussions/361

<div type='discussions-op-text'>

<sup>Originally posted by **Chralt98** May 23, 2022</sup>
[This](https://github.com/KILTprotocol/kilt-node/blob/develop/runtimes/spiritnet/src/lib.rs#L1216) will never get printed, if it relies on [frame-executive](https://docs.substrate.io/rustdocs/latest/src/frame_executive/lib.rs.html#265-272). Because the first unwrap in `try_runtime_upgrade()` would lead to a panic, therefore the code stops executing and wouldn't reach the `log::info` from the call stack.

```rust
#![allow(unused)]
fn main() {
  try_runtime_upgrade().map_err(|err| {
      println!("Additional information");
      err
  }).unwrap();
}

fn try_runtime_upgrade() -> Result<(), &'static str> {
    let x: Result<(), &'static str> = Err("fatal error").unwrap();
    Ok(())
}
```

```text
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: "fatal error"', src/main.rs:7:58
```
"Additional information" never mentioned.</div>

**Comment by shawntabrizi** [comment]
You shouldn't be using `unwrap`, you should be handling the result with something like `?` or similar.

Of course if you `unwrap` and cause a panic, you will not trigger proper Rust error handling.

**Comment by Chralt98** [comment]
My question: Isn't [this code ](https://github.com/KILTprotocol/kilt-node/blob/develop/runtimes/spiritnet/src/lib.rs#L1216) unreachable?

**Comment by weichweich** [comment]
Yes that code is unreachable. I think at time of writing I didn't expect that `Executive::try_runtime_upgrade()` panics if the try runtime encountered an error.
